container_workflow_tool/main.py

```diff
@@ -285,20 +286,29 @@ def _build_images(self, image_set, custom_args=[], branches=[]):
             self.logger.debug("Looping over all running builds")
             for proc, image in procs:
                 out = err = None
+                component = image["component"]
                 try:
                     self.logger.debug("Waiting {} seconds for {}".format(timeout,
-                                                                         image))
+                                                                         component))
                     out, err = proc.communicate(timeout=timeout)
                 except subprocess.TimeoutExpired:
                     msg = "{} not yet finished, checking next build"
-                    self.logger.debug(msg.format(image))
+                    self.logger.debug(msg.format(component))
                     continue
-
-                self.logger.info("{} build has finished".format(image))
+                if proc.returncode != 0:
+                    self.logger.error("{} build has failed".format(component))
+                else:
+                    self.logger.info("{} build has finished".format(component))
+                    # If this image triggers a new layer, build it
+                    if "trigger" in image:
+                        msg = "Triggering layered builds on image {}..."
+                        self.logger.info(msg.format(component))
+                        self.build_images(image["trigger"])
```

Comment:
Would it be possible to use only one function like `self.build_images` which will contain also `self._build_images` function? This is a bit misleading. `self._build_images` look like a property.

Reply:
Good question. I checked and in cwt we only use it in the `build_images` method. In our internal tooling that uses cwt as a library however, we add some additional stuff to it.
While it would be doable to add everything directly into `build_images`, I do not see much value to changing it.

Note that in Python a leading underscore in a name is not a naming convention only for properties, but for a convention for "private" attributes/methods in general.